### PR TITLE
8.0 cleanup and restructure

### PR DESCRIPTION
**Pull request by alexis-via**
Several enhancements:
1) re-organise the code to prepare the arrival of a new module to connect to a telephony system other than asterisk
2) move the wizard "Create CRM phonecall" from asterisk_click2dial_crm to crm_phone
3) Strip more characters from phone numbers sent to Asterisk and move that code to base_phone
4) small code cleanup + some move to new API

### CODE REVIEW

**Comment by brother-bernard** [comment]
:+1:

**Comment by chafique-delli** [comment]
:+1:

**Comment by sysadminmatmoz** [comment]
Before the merge cell phones calls worked fine, land phones had the issue where it sent (05)2222222 instead of +38652222222. Now I get this:

File "/opt/openerp/odoo/**other**/connector-telephony/asterisk_click2dial/asterisk_click2dial.py", line 407, in click2dial
    ast_number = self.convert_to_dial_number(erp_number)
  File "/opt/openerp/odoo/**base**/odoo_8/openerp/api.py", line 241, in wrapper
    return old_api(self, _args, *_kwargs)
TypeError: old_api() takes at least 3 arguments (2 given)

Am I missing something?

File: asterisk_click2dial/asterisk_click2dial.py [hunk 1]
```diff
@@ -429,15 +397,19 @@ class phone_common(orm.AbstractModel):
...
+            raise orm.except_orm(
                 _('Error:'),
                 _('Missing phone number'))
 
         user, ast_server, ast_manager = \
             self.pool['asterisk.server']._connect_to_asterisk(
                 cr, uid, context=context)
-        ast_number = self.pool['asterisk.server']._reformat_number(
-            cr, uid, erp_number, ast_server, context=context)
+        ast_number = self.convert_to_dial_number(erp_number)
```
Comment (yvaucher) [inline asterisk_click2dial/asterisk_click2dial.py:407]: self is not a record set

Either you need to convert this function to api8 or you can add cr, uid, context

Comment (sysadminmatmoz) [inline asterisk_click2dial/asterisk_click2dial.py:407]: Agree.
-                ast_number = self.convert_to_dial_number(cr, uid, erp_number, context=context)

solves my problem.

**Comment by alexis-via** [comment]
@sysadminmatmoz Sorry for that. The fix is in this PR : #34

**Comment by sysadminmatmoz** [comment]
I have to say thank you and you don't need to say sorry. We're doing a community work.

I did a real-life click2dial too - implemented it pro-bono in my hometown college (where I used to go 25 years ago), and with 400+ users they're a fantastic testing/development platform with their feedback.

**Comment by alexis-via** [comment]
@sysadminmatmoz Happy to hear that ! If you have more ideas on the asterisk-odoo connector, don't hesitate to share them with me and the community.

**Pull request merged**